src/core/func-core-tools.ts

```diff
@@ -0,0 +1,217 @@
+import os from 'os';
+import fs from 'fs';
+import path from 'path';
+import process from 'process';
+import { exec } from 'child_process';
+import { promisify } from 'util';
+import fetch from 'node-fetch';
+import unzipper from 'unzipper';
+import cliProgress from 'cli-progress';
+import { logger } from "./utils/logger";
+
+const RELEASES_FEED_URL = 'https://functionscdn.azureedge.net/public/cli-feed-v4.json';
+const DEFAULT_FUNC_BINARY = 'func';
+const VERSION_FILE = '.release-version';
+const LOCAL_CORE_TOOLS_FOLDER = '.swa/core-tools';
+const NODE_MAJOR_VERSION = getMajorVersion(process.versions.node);
+
+interface CoreToolsRelease {
+  version: string;
+  url: string;
+}
+
+interface CoreToolsZipInfo {
+  OS: string;
+  Architecture: string,
+  downloadLink: string;
+  size: string;
+}
```

Comment:
Can we move these types to https://github.com/Azure/static-web-apps-cli/blob/main/src/swa.d.ts?